students/srepking/Lesson06/mailroom4.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+def more_choices():
+    while True:
+        name = input('\nChoose an Option: \n'
+                     'e - to exit\n'
+                     'list -To see a list of names, or\n'
+                     'Type a name to start your thank you letter >>')
+        if name == 'e':
+            return
+        if name == 'list':
+            create_list(donors)
+        else:
+            print('\n''Ok, you want to write a letter for {}, '
+                  'lets see what we can do.'.format(name))
+
+            if donors.get(name) is None:
+                yes_no = input('The name you entered is not in the database.'
+                               'Would you like to add this name? y or n >>')
+                if yes_no == 'n':
+                    return
+
+            amount = input('\n''What is the donation amount? or \'e\' to exit >')
+            if amount == 'e':
+                return
+            else:
+                thank_you(donors, name, amount)
+
+
+
+def create_list(some_donors):
```

Comment:
no reason to pass in donors dict when it's already in global space.

Reply:
ah I see why! it's because you define `donors` in `if __name__ == "__main__":` block and when you import the mailroom that block isn't imported or visible.

if you define `donors` at the top of the file then you should have no problems.

Reply:
Great! I will try that. Thanks!